src/components/GamePanel.vue

```diff
@@ -88,27 +91,50 @@ function manualReset() {
     canFlipCards.value = true;
     clearTimeout(resetTimeout);
     allowReset = false;
+    if (firstCard !== undefined && secondCard !== undefined) {
+      firstCard.selection = CardSelection.UNSELECTED;
+      secondCard.selection = CardSelection.UNSELECTED;
+    }
+    firstCard = undefined;
+    secondCard = undefined;
   }
 }
 
 function resetCards() {
   canFlipCards.value = false;
   allowReset = true;
+  if (firstCard !== undefined && secondCard !== undefined) {
+    firstCard.selection = CardSelection.MISMATCH;
+    secondCard.selection = CardSelection.MISMATCH;
+  }
   resetTimeout = setTimeout(() => {
     openCardCount = 0;
     canFlipCards.value = true;
     allowReset = false;
+    if (firstCard !== undefined && secondCard !== undefined) {
+      firstCard.selection = CardSelection.UNSELECTED;
+      secondCard.selection = CardSelection.UNSELECTED;
+    }
+    firstCard = undefined;
+    secondCard = undefined;
   }, 5000);
 }
 
 function addPairToSummary(card1: CardData, card2: CardData) {
   canFlipCards.value = false;
+  card1.selection = CardSelection.MATCH;
+  card2.selection = CardSelection.MATCH;
   setTimeout(() => {
     foundPairs.value.push(new CardPair(card1, card2));
     card1.found = true;
     card2.found = true;
     openCardCount = 0;
+    card1.selection = CardSelection.UNSELECTED;
+    card2.selection = CardSelection.UNSELECTED;
     canFlipCards.value = true;
+    if (foundPairs.value.length == 6) {
+      isFinished.value = true;
+    }
```

Comment:
Maybe you can think about making isFinished a computed variable whether foundPairs length is 0 or not

Reply:
Ah, and I see it now, magic variables. You definitely have to change this.